Didi/add vs input controls (#1082)
telerik/maui-docs: controls/autocomplete/visual-structure.md

---
title: Visual Structure
page_title: .NET MAUI AutoComplete Documentation - AutoComplete Visual Structure
description: Learn more about the visual elements used in the Telerik UI for .NET MAUI AutoComplete control.
position: 0
slug: autocomplete-visual-structure
---

# .NET MAUI AutoComplete Visual Structure

Here are described all visual elements used in the [.NET MAUI AutoComplete]({%slug autocomplete-overview%}).

## AutoComplete Visual Structure

![Telerik UI for .NET MAUI AutoComplete Visual Structure](images/autocomplete-placeholder.png "Visual elements of AutoComplete control")

- `Placeholder`&mdash;The text which is used to give guidance to the end user on what can be entered/searched in the input.

## Tokens Structure

![Telerik UI for .NET MAUI AutoComplete Tokens Visual Structure](images/autocomplete-tokens-template.png "Visual elements of AutoComplete control")

- `Tokens`&mdash;When multiple items are selected from the dropdown list, these items appear as tokens. They can be deselected using their close button.

> - `ShowMore Template`&mdash;Represents an UI when the control is not focused and the space is not enough to show all tokens in a multiple selection scenario.

Reviewer: ```suggestion
- `ShowMore Template`&mdash;This template represents a UI that is displayed when the control is not focused and the space is insufficient to show all tokens in a multiple selection scenario.
```

## Filtering Structure

![Telerik UI for .NET MAUI AutoComplete Filtering Visual Structure](images/autocomplete-focused-text.png "Visual elements of AutoComplete control")

> - `Text`&mdash;The text written in the control's input filed.

Reviewer: ```suggestion
- `Text`&mdash;The text entered in the control's input filed.
```

> - [`RadTextInput`]({%slug entry-textinput%})&mdash;The control used for the text input

Reviewer: ```suggestion
- [`RadTextInput`]({%slug entry-textinput%})&mdash;The control used for the text input.
```

> - `Clear Button`&mdash;Clears the selection of the control both multiple and single.

Reviewer: ```suggestion
- `Clear Button`&mdash;Clears the selection in the control (for both multiple and single selection scenarios).
```
